app/components/PlaylistCover.tsx

```diff
@@ -0,0 +1,28 @@
+import { Song } from '@app/models/music'
+import CoverArt from '@app/components/CoverArt'
+import React from 'react'
+import { StyleSheet, ImageResizeMode, FlatList } from 'react-native'
+
+export function genPlaylistCover2x2Component(_songs: Song[]) {
+  const component: React.FC<{style: any, resizeMode: ImageResizeMode}> = ({style}) => {
+    const nestedStyle = { ...style, ...{ width: style.width / 2, height: style.height / 2, marginRight: 0 } }
+    // we need to wrap the songs in a custom object because the list will ignore null values
+    const songs = [...Array(4)].map((_, i) => ({s: i < _songs.length ? _songs[i] : null}))
+    return (
+      <FlatList
```

Comment:
`FlatList` is meant built more for long, scrolling lists where there are items in it that will be offscreen and not rendered, to optimize for performance.  In this case since this is a small component of fixed size a simple `View` would be more appropriate here.